labop/primitive_execution.py

```diff
@@ -190,19 +190,34 @@ def plate_coordinates_compute_output(self, inputs, parameter, sample_format):
 
 def measure_absorbance_compute_output(self, inputs, parameter, sample_format):
     if parameter.name == "measurements" and \
-       parameter.type == 'http://bioprotocols.org/labop#SampleData':
+       parameter.type == 'http://bioprotocols.org/labop#Dataset':
         input_map = input_parameter_map(inputs)
         samples = input_map["samples"]
-        sample_data = labop.SampleData(from_samples=samples)
+
+        # FIXME Bryan create the metadata.descriptions below from the input pins
+        sample_data = labop.Dataset(data=labop.SampleData(from_samples=samples), metadata=labop.SampleMetadata(descriptions="", for_samples=samples))
         return sample_data
 
+def attach_metadata_compute_output(self, inputs, parameter, sample_format):
```

Comment:
Tests are needed for this new functionality (and the other new functionalities)

Reply:
Yes.  Working on those in `test/test_sampledata.py`